core/common/src/MapLayerSettings.ts

```diff
@@ -150,38 +175,116 @@ export interface MapLayerKey {
+    return {
+      name: undefined !== changedProps.name ? changedProps.name : this.name,
+      visible: undefined !== changedProps.visible ? changedProps.visible : this.visible,
+      transparency: undefined !== changedProps.transparency ? changedProps.transparency : this.transparency,
+      transparentBackground: undefined !== changedProps.transparentBackground ? changedProps.transparentBackground : this.transparentBackground,
+    };
+  }
+
+  /** @internal */
+  public displayMatches(other: MapLayerSettings): boolean {
+    if (this.name !== other.name
+      || this.visible !== other.visible
+      || this.transparency !== other.transparency
+      || this.transparentBackground !== other.transparentBackground) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /** Return a unique string identifying the layers source... The URL for image layer or modelID for model layer  */
+  public abstract get source(): string;
+
+  /** @internal */
+  public matchesNameAndSource(name: string, source: string): boolean {
+    return this.name === name && this.source === source;
+  }
+}
+
+/** Normalized representation of a [[ImageMapLayerProps]] for which values have been validated and default values have been applied where explicit values not defined.
+ * Image map layers are created from servers that produce images that represent map tiles.  Map layers map also be represented by models.
+ * One or more map layers may be included within [[MapImagerySettings]] object.
+ * @see [[MapImagerySettings]]
+ * @see [[ModelMapLayerSettings]] for model based map layer settings.
+ * @beta
+ */
+export class ImageMapLayerSettings extends MapLayerSettings {
+  public readonly formatId: string;
+  public readonly url: string;
   public userName?: string;
   public password?: string;
   public accessKey?: MapLayerKey;
-
-  public setCredentials(userName?: string, password?: string) {
-    this.userName = userName;
-    this.password = password;
-  }
+  public readonly isBase: boolean;
+  public readonly subLayers: MapSubLayerSettings[];
+  public override get source(): string { return this.url; }
 
   /** @internal */
   protected constructor(url: string, name: string, formatId: string, visible = true,
     jsonSubLayers: MapSubLayerProps[] | undefined = undefined, transparency: number = 0,
     transparentBackground = true, isBase = false, userName?: string, password?: string, accessKey?: MapLayerKey) {
+    super(name, visible, transparency, transparentBackground);
     this.formatId = formatId;
-    this.name = name;
-    this.visible = visible;
-    this.transparentBackground = transparentBackground;
     this.isBase = isBase;
+    this.url = url;
+    this.userName = userName;
+    this.password = password;
+    this.accessKey = accessKey;
     this.subLayers = new Array<MapSubLayerSettings>();
     if (jsonSubLayers !== undefined) {
       let hasUnnamedGroups = false;
```

Comment:
Can I delete this `hasUnnamedGroups` stuff or is it supposed to be used for something?

Reply:
Yes.... I believe so - it certainly doesn't seem to do anything.   @mdastous-bentley ?

Reply:
yes, I believe its left over from a previous implementation.